### PR TITLE
[torch.package/TorchScript] flag to gate allowance of TS serializaiton in torch.package

### PR DESCRIPTION
**Pull request by lobstergrindset**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57888 [PyTorch Edge] bytecode version bump to v5 and enable share constant table
* #58182 [PyTorch Edge] share tensors in mobile with new api
* **#57678 [torch.package/TorchScript] flag to gate allowance of TS serializaiton in torch.package**
* #54894 [torch.Package/TorchScript] TS into torch.Package test cases
* #57573 [torch.package/TorchScript] logic to enable sharing of tensors on load
* #54893 [torch.Package/TorchScript] torch.Package python logic to save TorchScript
* #56299 [torch.Package/TorchScript] ScriptModuleSerializer add unified format
* #55958 [torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter
* #54891 [torch.Package/TorchScript] TS serialization importer to handle unified format

Differential Revision: [D28232891](https://our.internmc.facebook.com/intern/diff/D28232891)

### CODE REVIEW

**Comment by Lilyjjo** [comment]
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28232891).

**Comment by Lilyjjo** [comment]
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28232891).

**Comment by Lilyjjo** [comment]
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28232891).

File: torch/package/package_exporter.py [hunk 1]
```diff
@@ -112,6 +112,7 @@ def __init__(
         self.provided: Dict[str, bool] = {}
         self.verbose = verbose
         self.script_module_serializer = torch._C.ScriptModuleSerializer(self.zip_file)
+        self.gate_torchscript_serialization = True
```
Comment (suo) [inline torch/package/package_exporter.py:115]: let's just make this a global with a leading underscore, and set it in the `setUp()` for the tests. It's fine to be a universal setting for now.

**Review by suo — CHANGES_REQUESTED** [review]

**Review by suo — APPROVED** [review]

**Comment by Lilyjjo** [comment]
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28232891).

**Comment by Lilyjjo** [comment]
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28232891).

**Comment by Lilyjjo** [comment]
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28232891).

**Comment by Lilyjjo** [comment]
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28232891).

**Pull request closed without merging**